Client/Frontend/Browser/Playlist/PlaylistToast.swift

```diff
@@ -11,35 +11,94 @@ import SnapKit
-    }
+private class ToastShadowView: UIView {
+    private var shadowLayer: CAShapeLayer?
     
-    private class HighlightableButton: UIButton {
-        override var isHighlighted: Bool {
-            didSet {
-                backgroundColor = isHighlighted ? .white : .clear
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        let path = UIBezierPath(roundedRect: bounds,
+                                cornerRadius: ButtonToastUX.toastButtonBorderRadius).cgPath
+        if let shadowLayer = shadowLayer {
+            shadowLayer.path = path
+            shadowLayer.shadowPath = path
+        } else {
+            shadowLayer = CAShapeLayer().then {
+                $0.path = path
+                $0.fillColor = UIColor.clear.cgColor
+                $0.shadowColor = UIColor.black.cgColor
+                $0.shadowPath = path
+                $0.shadowOffset = CGSize(width: 0.0, height: 0.0)
+                $0.shadowOpacity = 0.5
+                $0.shadowRadius = ButtonToastUX.toastButtonBorderRadius
+            }
+            
+            shadowLayer?.do {
+                layer.insertSublayer($0, at: 0)
             }
         }
     }
+}
+
+private class HighlightableButton: UIButton {
+    private var shadowLayer: CAShapeLayer?
     
-    private let shadowLayer = CAShapeLayer().then {
-        $0.fillColor = nil
-        $0.shadowColor = UIColor.black.cgColor
-        $0.shadowOffset = CGSize(width: 2.0, height: 2.0)
-        $0.shadowOpacity = 0.15
-        $0.shadowRadius = ButtonToastUX.toastButtonBorderRadius
+    var shadowLayerZOrder: Int {
+        guard let shadowLayer = shadowLayer else { return -1 }
+        return self.layer.sublayers?.firstIndex(of: shadowLayer) ?? -1
+    }
+    
+    var isShadowHidden: Bool = false {
+        didSet {
+            shadowLayer?.isHidden = isShadowHidden
+        }
     }
     
+    override var isHighlighted: Bool {
+        didSet {
+            backgroundColor = isHighlighted ? UIColor.white.withAlphaComponent(0.2) : .clear
+        }
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        let path = UIBezierPath(roundedRect: bounds,
+                                cornerRadius: ButtonToastUX.toastButtonBorderRadius).cgPath
+        if let shadowLayer = shadowLayer {
+            shadowLayer.path = path
+            shadowLayer.shadowPath = path
+        } else {
+            shadowLayer = CAShapeLayer().then {
+                $0.path = path
+                $0.fillColor = UIColor.clear.cgColor
+                $0.shadowColor = UIColor.black.cgColor
+                $0.shadowPath = path
+                $0.shadowOffset = CGSize(width: 0.0, height: 0.0)
+                $0.shadowOpacity = 0.5
+                $0.shadowRadius = ButtonToastUX.toastButtonBorderRadius
+            }
+            
+            shadowLayer?.do {
+                layer.insertSublayer($0, at: 0)
```

Comment:
This part is the same as the previous shadowlayer, no? Maybe abstract it a little